src/Core/Framework/Api/OAuth/AccessToken.php

```diff
@@ -79,21 +79,11 @@ public function addScope(ScopeEntityInterface $scope): void
 
     public function initJwtConfiguration(): void
     {
-        if ($this->privateKey instanceof FakeCryptKey) {
-            $this->jwtConfiguration = $this->privateKey->configuration;
-
-            return;
+        if (!$this->privateKey instanceof FakeCryptKey) {
```

Comment:
I don't really understand this code but this seems to me that we always have a `FakeCryptKey` - is that correct?

Reply:
Yes, this is basically just there to nail down the generic type from the trait from the library 

I tried now to nail down the type in the implementation, as PHP should allow to be more specific there

Reply:
this was already brought up here: https://github.com/shopware/shopware/pull/5986#discussion_r1901795082

Reply:
> I tried now to nail down the type in the implementation, as PHP should allow to be more specific there

Damn it doesn't work for traits, I'll bring this back